api/src/main/java/com/datastrato/gravitino/rel/types/Type.java

```diff
@@ -98,7 +98,19 @@ abstract class IntervalType extends PrimitiveType {}
   abstract class ComplexType implements Type {}
 
   /** The base type of all integral types. */
-  abstract class IntegralType extends NumericType {}
+  abstract class IntegralType extends NumericType {
+    private final boolean signed;
```

Comment:
I would suggest rename it to `unsigned` as data type like `integer` is signed by default, and the default value of bool type is `false`.

Reply:
The current implementation is LGTM. `signed` is in the constructor and is more intuitive